identity/src/lib/db/queries.rs

```diff
@@ -0,0 +1,88 @@
+use diesel::prelude::*;
+use diesel::result::QueryResult;
+
+use super::models::*;
+use super::schema;
+use super::DbConn;
+
+pub fn create_user(user: UserAddUpdate, db: &DbConn) -> QueryResult<User> {
+    use schema::users::dsl::users;
+
+    diesel::insert_into(users).values(&user).get_result(db)
+}
+
+pub fn get_user(user_id: i32, db: &DbConn) -> QueryResult<User> {
+    use schema::users::dsl::users;
+
+    users.find(user_id).first(db)
+}
+
+pub fn list_users(
+    offset: i64,
+    limit: i64,
+    user_active: Option<bool>,
+    db: &DbConn,
+) -> QueryResult<Vec<User>> {
+    use schema::users::dsl::*;
+
+    match user_active {
+        Some(val) => users
+            .filter(active.eq(val))
+            .offset(offset)
+            .limit(limit)
+            .load(db),
+        None => users.offset(offset).limit(limit).load(db),
+    }
+}
+
+pub fn update_user(user: User, db: &DbConn) -> QueryResult<User> {
+    use schema::users::dsl::*;
+
+    diesel::update(users.find(user.id as i32))
+        .set(active.eq(user.active))
+        .get_result(db)
```

Comment:
Sometimes you use `.get_result` and sometimes `.load` even though they are just aliases.

Reply:
The `.get_result` and `.load` are not aliases, but `.load` and `.get_results` are. I could replace the `.load` with `.get_results`, but I do not really see the value of this right now.

Reply:
> The `.get_result` and `.load` are not aliases, but `.load` and `.get_results` are. I could replace the `.load` with `.get_results`, but I do not really see the value of this right now.

Ups, then I confused that 😓